File: java/3248. Snake in Matrix.java
class Solution {
    public int finalPositionOfSnake(int n, List<String> commands) {
        int x = 0;
        int y = 0;
        for (String s : commands) {
            switch(s) {
                case "LEFT":
                    y--;
                    break;
                case "RIGHT":
                    y++;
                    break;
                case "UP":
                    x--;
                    break;
                default:
                    x++;
                    break;
            }
        }
        return x*n + y;
    }
}

File: go/3248. Snake in Matrix.go
func finalPositionOfSnake(n int, commands []string) int {
    x, y := 0, 0
    for _, c := range commands {
		switch c {
		case "LEFT":
			y--
		case "RIGHT":
			y++
		case "UP":
			x--
		default:
			x++
		}
    }
	return x*n + y
}